polars/polars-core/src/chunked_array/kernels/take_agg.rs

```diff
@@ -1,19 +1,19 @@
 //! kernels that combine take and aggregations.
 use crate::prelude::*;
 use arrow::array::{Array, PrimitiveArray};
-use polars_arrow::buffer::IsValid;
+use arrow::types::NativeType;
 
 /// Take kernel for single chunk without nulls and an iterator as index.
 pub(crate) unsafe fn take_agg_no_null_primitive_iter_unchecked<
-    T: PolarsNumericType,
+    T: NativeType,
     I: IntoIterator<Item = usize>,
-    F: Fn(T::Native, T::Native) -> T::Native,
+    F: Fn(T, T) -> T,
 >(
     arr: &PrimitiveArray<T>,
     indices: I,
     f: F,
-    init: T::Native,
-) -> T::Native {
+    init: T,
+) -> T {
```

Comment:
Arrow2 does not have `::Native`; there is one trait, `NativeType`, that is implemented for all arrow primitive types with a corresponding rust type (`u32`, `i64`, etc.). A bit easier to write.

Reply:
Definitely! But how would you call this with for example `Date32`? First coercing to physical?

Reply:
Yes, e.g. [here](https://github.com/jorgecarleitao/arrow2/blob/main/src/compute/take/mod.rs#L71) all logical types backed by the same physical type get the same branch of the match. This is true for all operations whose semantics is independent of the logical type (like take, filter, is_null, drop_nan, concat, etc).